sympy/solvers/solveset.py

```diff
@@ -811,6 +811,46 @@ def solveset_complex(f, symbol):
     return solveset(f, symbol, S.Complexes)
 
 
+def solvify(f, symbol, domain=S.Reals):
```

Comment:
Giving it a second thought, we need the function `f` to try and compute its period (via `periodicity`).
This function is true to its cause even if we use `solvify(f, symbol, domain)`(by internalising the call to `solveset`). 
So, I think that using the current implementation is a good idea.